decoder/decoder.go

```diff
@@ -0,0 +1,16 @@
+package decoder
+
+import (
+	"net/http"
+)
+
+// Decoder decoder.
+//
+//go:generate mockery --name=Decoder --outpkg=mockdecoder --output=./mockdecoder
+type Decoder interface {
+	ContentType() string
+	Decode(r *http.Request, ptr any) error
+}
+
+// Decoders decoders.
```

Comment:
// Decoders is a map of decoders where keys are content types for given decoders.